styx-service-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -220,16 +245,19 @@ WorkflowInstanceExecutionData executionData(WorkflowInstance workflowInstance) t
 
   private List<WorkflowInstanceExecutionData> executionData(
       Set<WorkflowInstance> workflowInstancesSet) {
-    return workflowInstancesSet.parallelStream()
-        .map(workflowInstance -> {
+    return workflowInstancesSet.stream()
+        .map(workflowInstance -> asyncIO(() -> {
```

Comment:
we don't need to catch the `IOException` because the lambda would be mapped to a `IOOperation` and those can throw `IOException`